### ISSUE TITLE
ExternalPerturbationLangevinIntegrator cannot reset()

### ISSUE THREAD

**Issue by pgrinaway** [choderalab/openmmtools#381]
It seems that a differently-named variable (`first_step` vs `step`) is the culprit. Reproducing the problem is straightforward:

```python
from openmmtools.integrators import ExternalPerturbationLangevinIntegrator
i  =  ExternalPerturbationLangevinIntegrator()
i.reset()
```

and you get:
`Exception: Illegal global variable name: step`. I think changing `first_step` to `step` should resolve this.

**Comment by jchodera** [comment]
@pgrinaway This is the cause of your first step having crazy work contributions. I'd just change `first_step` to `step` and have each integrator step increment `step`.

**Comment by pgrinaway** [comment]
Resolved in #382
